### PR TITLE
chore: migrate to node:test with c8

### PR DESCRIPTION
**Pull request by narendra-reddy-nf**
This PR adds [borp](https://github.com/mcollina/borp) and removes `tap`

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by simoneb — APPROVED** [review]
LGTM

File: tests/index.test.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import Fastify from 'fastify'
 import fastifySession from '@fastify/secure-session'
 import querystring from 'querystring'
 import fastifyFlash from '../src'
 
-const key = readFileSync(join(__dirname, '..', 'secret-key'))
+const key = readFileSync(join(__dirname, '..', '..', 'secret-key'))
 
-test('should set error message and and clear up after displaying.', async t => {
+test('should set error message and and clear up after displaying.', async (t:TestContext) => {
```
Comment (simoneb) [inline tests/index.test.ts:11]: let's always include a space after the colon. not sure why the linting is not picking this up

```suggestion
test('should set error message and and clear up after displaying.', async (t: TestContext) => {
```

File: package.json [hunk 2]
```diff
@@ -14,9 +14,9 @@
     "update": "npx npm-check -u",
     "prepublishOnly": "npm run build",
     "pretest": "npx -p \"@fastify/secure-session\" secure-session > secret-key",
-    "test": "tap",
+    "test": "borp --coverage",
     "test:report": "npm run test | tee out.tap",
-    "test:reporter": "tap-mocha-reporter xunit < out.tap > junit-testresults.xml",
+    "test:reporter": "npm run test --reporter=junit>junit-testresults.xml",
```
Comment (mcollina) [inline package.json:19]: where is this used?

File: tests/index.test.ts (hunk 1 above)
Comment (climba03003) [inline tests/index.test.ts:11]: Why it requires explicit types?

File: package.json (hunk 2 above)
Comment (narendra-reddy-nf) [reply package.json:19]: It's not being used in the existing code.

File: tests/index.test.ts (hunk 1 above)
Comment (narendra-reddy-nf) [reply tests/index.test.ts:11]: As we are using `t.assert` for assertions, it expects explicit type annotation.

**Review by gurgunday — APPROVED** [review]

**Review by simoneb — APPROVED** [review]
LGTM

File: tests/index.test.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 import Fastify from 'fastify'
 import fastifySession from '@fastify/secure-session'
 import querystring from 'querystring'
 import fastifyFlash from '../src'
 
-const key = readFileSync(join(__dirname, '..', 'secret-key'))
+const key = readFileSync(join(__dirname, '..', '..', 'secret-key'))
 
-test('should set error message and and clear up after displaying.', async t => {
+test('should set error message and and clear up after displaying.', async (t: TestContext) => {
```
Comment (simoneb) [inline tests/index.test.ts:11]: why does the t variable need to be explicitly typed?

Comment (climba03003) [reply tests/index.test.ts:11]: Same question as mine and the answer https://github.com/fastify/fastify-flash/pull/110#discussion_r1756199811

File: tests/index.test.ts (hunk 1 above)
Comment (simoneb) [reply tests/index.test.ts]: interesting, why would Node not know that the callback function to `test` is of type `TestContext`, do you know?

Comment (climba03003) [reply tests/index.test.ts]: It is actually the problem of TypeScript.
Using `assert` keyword will have some limitation that requires explicit `this`.

Comment (climba03003) [reply tests/index.test.ts]: Long story if you interested https://github.com/microsoft/TypeScript/issues/36931
Short Cap on https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70193#discussion_r1701867726

Comment (simoneb) [reply tests/index.test.ts]: Thanks @climba03003 . So we're happy with the explicit typing then?

Comment (climba03003) [reply tests/index.test.ts]: Yes

**Pull request merged**